sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -270,12 +273,25 @@ trait CheckAnalysis extends PredicateHelper {
 
         operator match {
           case o if o.children.nonEmpty && o.missingInput.nonEmpty =>
+            val resolver = conf.resolver
+            val commonAttrs = o.missingInput.filter(x =>
+              o.inputSet.exists(y => resolver(x.name, y.name)))
             val missingAttributes = o.missingInput.mkString(",")
-            val input = o.inputSet.mkString(",")
+            val availableAttributes = o.inputSet.mkString(",")
+            val repeatedNameHint = if (commonAttrs.size > 0) {
+              val commonNames = commonAttrs.map(_.name).mkString(",")
+              s"""\n|Observe that attribute(s) $commonNames appear in two
```

Comment:
I'm not sure if this is always the case. Some failures in the query plan analysis can cause missing attributes, I mean.

Reply:
The new line `\n` is intentional?

Reply:
I'll rewrite the wording a bit so it hints to this as the cause (it's the most usual I suspect). The newline is for readability of the error message. I could remove it.